Dockerfile

```
@@ -0,0 +1,67 @@
+FROM python:3.9-alpine as base
+
+# Set environment variable
+ENV POETRY_HOME="/opt/poetry"
+ENV PATH="$POETRY_HOME/bin:$PATH"
+
+# Install poetry
+RUN pip install poetry Flask gunicorn
+
+# Set up the working directory
+WORKDIR /app
+
+# Copy the content of the repo inside the image
+COPY . /app
+
+
+
+# Dependecies stage
+FROM base as dependencies
+
+# Install dependecies withtout the dev dependencies
+RUN poetry install --no-dev
+
+
+# Development stage
+FROM dependencies as development_env
+
+# Install the development dependencies
+RUN poetry install 
+
+# Expose the Flask development port
+EXPOSE 5000
+
+
+# Production environment
+FROM base as production
+
+#copy the dependencies
+COPY --from=dependencies /usr/local/lib/python3.9/site-packages /usr/local/lib/python3.9/site-packages
```

Here, we might as well just build production directly on top of the `dependencies` stage:
```suggestion
FROM dependencies as production

```

There's no other outputs from dependencies that I think we're really suffering from